### PR TITLE
Elevate the removal of shortcuts the current user cannot delete

### PR DESCRIPTION
**Pull request by GabrielDuf**
This pull request introduces a new batch shortcut deletion feature, improves the safety and atomicity of shortcut removal, and adds comprehensive tests for shortcut deletion logic. The changes refactor both Desktop and Start Menu shortcut deletion to use a new `ShortcutFileRemover` utility, update the CLI to support batch shortcut deletion (including elevation), and extend test coverage for these scenarios.

**Shortcut Deletion Refactoring and Batch Support:**

* Both `DesktopShortcutsDatabase` and `StartMenuShortcutsDatabase` now use the new `ShortcutFileRemover` utility for deleting shortcuts, supporting batch deletion and improving error handling and logging. Methods for deleting multiple shortcuts at once were added and refactored to use this utility. [[1]](diffhunk://#diff-d4ddcd6a96b9c7616163ea5b0b1dd8b78c841275935f1f75d9d4e959f6c56240L83-R92) [[2]](diffhunk://#diff-d4ddcd6a96b9c7616163ea5b0b1dd8b78c841275935f1f75d9d4e959f6c56240R207) [[3]](diffhunk://#diff-d4ddcd6a96b9c7616163ea5b0b1dd8b78c841275935f1f75d9d4e959f6c56240L219-R220) [[4]](diffhunk://#diff-d4ddcd6a96b9c7616163ea5b0b1dd8b78c841275935f1f75d9d4e959f6c56240L236-R237) [[5]](diffhunk://#diff-d4ddcd6a96b9c7616163ea5b0b1dd8b78c841275935f1f75d9d4e959f6c56240R250-R252) [[6]](diffhunk://#diff-eba9756492b5f9206a3557b70ef29f6d75d19605edbcac0f507f00cd8a15daa0R4) [[7]](diffhunk://#diff-eba9756492b5f9206a3557b70ef29f6d75d19605edbcac0f507f00cd8a15daa0R613-R615) [[8]](diffhunk://#diff-eba9756492b5f9206a3557b70ef29f6d75d19605edbcac0f507f00cd8a15daa0L633-R637) [[9]](diffhunk://#diff-eba9756492b5f9206a3557b70ef29f6d75d19605edbcac0f507f00cd8a15daa0L642-R646) [[10]](diffhunk://#diff-eba9756492b5f9206a3557b70ef29f6d75d19605edbcac0f507f00cd8a15daa0R694-R702) [[11]](diffhunk://#diff-eba9756492b5f9206a3557b70ef29f6d75d19605edbcac0f507f00cd8a15daa0L934-R960)

* The `ManageShortcutsViewModel` now batches shortcut deletions and only performs disk operations once per batch, improving efficiency and reducing the risk of partial deletions. [[1]](diffhunk://#diff-aca7ce1f3da3334a81e60c473898af0569569ccba1ce37cb92751960864d3c53R277-R279) [[2]](diffhunk://#diff-aca7ce1f3da3334a81e60c473898af0569569ccba1ce37cb92751960864d3c53L296-R299) [[3]](diffhunk://#diff-aca7ce1f3da3334a81e60c473898af0569569ccba1ce37cb92751960864d3c53L314-R317) [[4]](diffhunk://#diff-aca7ce1f3da3334a81e60c473898af0569569ccba1ce37cb92751960864d3c53L332-R349)

**CLI and Command Dispatcher Enhancements:**

* Adds a new CLI argument `--delete-shortcuts <path> [<path>...]` to delete shortcuts (with elevation if needed), and updates the dispatcher to handle this argument. Documentation is updated to describe this new flag and its restrictions for safety. [[1]](diffhunk://#diff-7340b8a6d2c5b52c935f75ec72038d28b8c606323ea41bd29ced4b521eae7761R261) [[2]](diffhunk://#diff-28d0b4bbbd9fb6d9843f75dcebcd23f75066dd108724f08483d06f963f43e101R42) [[3]](diffhunk://#diff-28d0b4bbbd9fb6d9843f75dcebcd23f75066dd108724f08483d06f963f43e101R124-R128) [[4]](diffhunk://#diff-28d0b4bbbd9fb6d9843f75dcebcd23f75066dd108724f08483d06f963f43e101R335-R359)

**Testing Improvements:**

* Introduces a comprehensive test suite for `ShortcutFileRemover`, covering edge cases (e.g., read-only files, files in use, invalid paths, wildcards, alternate data streams, batch operations, and elevated instance behavior).

**Codebase Consistency:**

* Updates using directives in affected files to include the new `UniGetUI.Core.Tools` namespace, ensuring all shortcut deletion logic is centralized and consistent. [[1]](diffhunk://#diff-d4ddcd6a96b9c7616163ea5b0b1dd8b78c841275935f1f75d9d4e959f6c56240R3) [[2]](diffhunk://#diff-eba9756492b5f9206a3557b70ef29f6d75d19605edbcac0f507f00cd8a15daa0R4) [[3]](diffhunk://#diff-73631f7224f53446fc00e825d922fb136c0d774cd32e012e5b4d198e17f2a793R5)

These changes improve the reliability, security, and maintainability of shortcut management throughout the application.

### CODE REVIEW

File: src/UniGetUI.Core.Tools/ShortcutFileRemover.cs [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+
+            if (
+                !fullPath.StartsWith(
+                    fullRoot + Path.DirectorySeparatorChar,
+                    StringComparison.OrdinalIgnoreCase
+                )
+            )
+                continue;
+
+            return !CrossesReparsePoint(fullRoot, fullPath);
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.Core.Tools/ShortcutFileRemover.cs]: [f94abb8](https://github.com/Devolutions/UniGetUI/pull/5372/commits/f94abb879f7985eaa7786b1aa82e306a085fb002)

File: src/UniGetUI.Core.Tools/ShortcutFileRemover.cs [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+        Environment.SpecialFolder.CommonDesktopDirectory,
+        Environment.SpecialFolder.Programs,
+        Environment.SpecialFolder.CommonPrograms,
+    ];
+
+    public static IReadOnlyList<string>? TEST_ShortcutRootsOverride;
+
+    public static bool Delete(string shortcutPath)
+    {
+        if (TryDelete(shortcutPath, out bool accessDenied))
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.Core.Tools/ShortcutFileRemover.cs:62]: [f94abb8](https://github.com/Devolutions/UniGetUI/pull/5372/commits/f94abb879f7985eaa7786b1aa82e306a085fb002)

File: src/UniGetUI.Core.Tools.Tests/ShortcutFileRemoverTests.cs [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    [Fact]
+    public void TheElevatedInstanceRefusesAShortcutReachedThroughAJunction()
+    {
+        if (!OperatingSystem.IsWindows())
+            return;
+
+        string outside = CreateFile(_outsideRoot, "Payload.lnk");
+        string junction = Path.Combine(_root, "Vendor");
+        if (!TryCreateJunction(junction, _outsideRoot))
+            return;
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.Core.Tools.Tests/ShortcutFileRemoverTests.cs]: [c6e77f5](https://github.com/Devolutions/UniGetUI/pull/5372/commits/c6e77f532f6762d70a8d0c592b482cf40a8a5521)
